### ISSUE TITLE
eclib: set0_{128,256} issue; Jasmin v2022.04.0

### ISSUE THREAD

**Issue by tfaoliveira** [jasmin-lang/jasmin#128]
This issue is described in the context of Jasmin `v2022.04.0` and EasyCrypt `r2022.04`.

When extracting to EasyCrypt a Jasmin function that contains `#set0_128()` or `#set0_256()`, those statements are extracted to `<- set0_128 ;` or `<- set0_256 ;`.

When checking the extracted file with EasyCrypt, I get the error:
```
This expression has type
   bool * bool * bool * bool * bool * W128.t
but is expected to have type
   W128.t
```

* `set0_XX` is defined here: https://github.com/jasmin-lang/jasmin/blob/7f67b5b166e237bbf54228286d4b2bbe3fb07c6b/eclib/JWord.ec#L1482
* Theory `W256` is defined here: https://github.com/jasmin-lang/jasmin/blob/7f67b5b166e237bbf54228286d4b2bbe3fb07c6b/eclib/JWord.ec#L2520

I think it is necessary to define `set0_128` and `set0_256` differently (without returning the flags) for these particular cases.

**Comment by fdupress** [comment]
Are there any uses of those instructions where the flags are not ignored in the Jasmin source? Instead of defining `set0_128` and `set0_256` differently, it's probably a better idea to tweak extraction to mimic the source's behaviour w.r.t. ignoring the flags.

**Comment by bgregoir** [comment]
Hi Tiago, this is clearly a bug in the extraction and the easycrypt model. I will fix this.

**Comment by eponier** [comment]
Before, `set0` for sizes 128 and 256 was called `setw0`. Maybe this is related to this bug.

**Comment by tfaoliveira** [comment]
For u128 and u256 flags are not affected. I think this is possible to see in `Definition Oset0_instr` from https://github.com/jasmin-lang/jasmin/blob/7906ac9f97afdb40c95d503e374c75f3f13ca836/proofs/compiler/x86_extra.v#L35, and I also double-checked in the Intel developer's manual. 

I would risk saying that the extraction is OK and only the `eclib` needs to be adjusted.

**Comment by vbgl** [comment]
This has been fixed in #134.

**Comment by tfaoliveira** [comment]
Tested on libjade; Everything fine :-)
